pkg/innerring/subnet.go

```diff
@@ -315,5 +319,93 @@ func (s *Server) handleSubnetRemoval(e event.Event) {
 		return
 	}
 
-	// TODO: #1162 handle removal of the subnet in netmap candidates
+	// handle subnet changes in netmap
+
+	candidates, err := s.netmapClient.GetCandidates()
+	if err != nil {
+		s.log.Error("getting netmap candidates",
+			zap.Error(err),
+		)
+
+		return
+	}
+
+	var removedID subnetid.ID
+	_ = removedID.Unmarshal(delEv.ID()) // parsing errors were checked above
+
+	for _, c := range candidates.Nodes {
+		s.processCandidate(delEv.TxHash(), removedID, c)
+	}
+}
+
+func (s *Server) processCandidate(txHash neogoutil.Uint256, removedID subnetid.ID, c netmap.Node) {
+	removeFromNM := true
+	removeSubnet := false
+	log := s.log.With(
+		zap.String(
+			"public_key",
+			hex.EncodeToString(c.NodeInfo.PublicKey()),
+		),
+		zap.String(
+			"removed_subnet",
+			removedID.String(),
+		),
+	)
+
+	err := c.NodeInfo.IterateSubnets(func(id subnetid.ID) error {
+		if removedID.Equals(&id) {
+			removeSubnet = true
+			return netmap.ErrRemoveSubnet
+		}
+
+		// not remove from netmap if there
+		// is at least one more subnet
+		removeFromNM = false
+
+		return nil
+	})
+	if err != nil {
+		s.log.Error("iterating node's subnets",
+			zap.Error(err),
+		)
+	}
+
+	// remove from netmap if there are no
+	// more subnets for the node or if it
+	// is impossible to process its subnet
+	// information
+	if err != nil || removeFromNM {
```

Comment:
Pay attention on `IterateSubnets` docs: it returns an error if node has no or has been removed from all subnets. So we can get rid of `removeFromNM`.

Note that in this case this `if` statement and the one above can be merged.

Reply:
yes, read the doc but missed "...not included in any subnet **by the end of the loop**", thanks, fixed

Reply:
oh no: i remembered why i did so: yes, i read the doc correctly but did not want to merge two cases (printing the same type of logs and so on): "could not read the subnet ID" (which seems an error) and "node does not have any more subnet " (which seems as ok situation)

on the other hand that is not that critical so maybe leave them merged, what do you think?

Reply:
`node does not have any more subnet` is expected to raise pretty rare, so I'd print it anyway.

Reply:
@cthulhu-rider so maybe we need to have public `ErrNoSubnet` in SDK